packages/server/lib/open_project.ts

```diff
@@ -186,6 +186,8 @@ export class OpenProject {
         return await browsers.connectToNewSpec(browser, { onInitializeNewBrowserTab, ...options }, automation)
```

Comment:
Do we need to `connectToNewSpec` as well to create a new instance if one doesn't exist? Currently the browser instance doesn't exist, `browsers.connectToNewSpec` will see this isn't defined and throw:
```js
  async connectToNewSpec (browser: Browser, options: BrowserNewTabOpts, automation: Automation) {
    debug('connecting to new chrome tab in existing instance with url and debugging port', { url: options.url })

    const browserCriClient = this._getBrowserCriClient()

    if (!browserCriClient) throw new Error('Missing browserCriClient in connectToNewSpec')
```

Reply:
connectToNewSpec isn't run because 'options.shouldLaunchNewTab' is false, so we fall into the relaunchBrowser flow

Reply:
When does this get reset to false?

Reply:
i assume in onError which end up calling 'onEarlyExit' in run.ts. Theres a lot of redirections.